src/render.rs

```diff
@@ -370,75 +371,123 @@ fn render_cpu_bars(
     f: &mut Frame<'_, ZBackend>,
     style: &Style,
 ) {
-    let mut cpus = app.cpus.to_owned();
-    let mut bars: Vec<(&str, u64)> = vec![];
-    let mut bar_gap: u16 = 1;
-
-    let mut np = cpus.len() as u16;
-    if np == 0 {
-        np = 1;
+    let cpus = app.cpus.to_owned();
+    if cpus.is_empty() {
+        return;
     }
 
-    let mut constraints: Vec<Constraint> = vec![];
-    let half = if np > width - 2 {
-        constraints.push(Constraint::Percentage(50));
-        constraints.push(Constraint::Percentage(50));
-        np as usize / 2
-    } else {
-        constraints.push(Constraint::Percentage(100));
-        np as usize
-    };
-    //compute bar width and gutter/gap
-
-    if width > 2 && (half * 2) >= (width - 2) as usize {
-        bar_gap = 0;
-    }
-    let width = width - 2;
-    let mut cpu_bw = ((width as f32 - (half as u16 * bar_gap) as f32) / half as f32) as i16;
-    if cpu_bw < 1 {
-        cpu_bw = 1;
-    }
-    let cpu_bw = cpu_bw as u16;
-    for (i, (p, u)) in cpus.iter_mut().enumerate() {
-        if i > 8 && cpu_bw == 1 {
-            p.remove(0);
-        }
-        bars.push((p.as_str(), *u));
-    }
+    let core_count = cpus.len() as u16;
+    let widest_label = cpus.iter().map(|(s, _)| s.len()).max().unwrap_or(0) as u16;
+    let full_width = widest_label * core_count + (core_count - 1);
 
     Block::default()
-        .title(format!("CPU(S) {}@{} MHz", np, app.frequency).as_str())
+        .title(
+            format!(
+                "CPU{} {}@{} MHz",
+                core_count,
```

Comment:
This line needs to be swapped with the one below it otherwise the core count ends up next to CPU instead of the "S"